### PR TITLE
Move debug pages under test folder

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist

I've moved the debug folder under the test folder and changed its name to `debug-pages`.
This way the main folder is a lot cleaner and debug pages are for tests anyway to I think it's better that they'll be there.
I also created a index.html generation file in order to ease the use of the debug pages.
It can be improved a lot, I know, but for now I think it's good enough.
The last thing which I'm thinking about moving is the rollup/build folders - they are practically the same concept and rollup is a specific technology which I don't think there should be a dedicated folder for.
But this is not part of this PR :-)

image

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Document any changes to public APIs.
 - [x] Post benchmark scores.
 - [x] Manually test the debug page.
 - [x] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [x] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Comment by birkskyum** [comment]
This is great :) The links in the debug index were broken for me, so I fixed the list of url's from pointing to test/example-file.html to test/debug-pages/example-file.html

File: test/debug-pages/index.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
+
```
Comment (wipfli) [inline test/debug-pages/index.html:1]: If that is a generated file, do not put it in source control.

**Review by wipfli — CHANGES_REQUESTED** [review]
Looks good thanks for doing this @HarelM. My only comment would be to not put a generated file in git...

File: test/debug-pages/index.html (hunk 1 above)
Comment (birkskyum) [reply test/debug-pages/index.html:1]: Good point - we can just add it to the codegen like the rest of the generated files.

File: build/generate-debug-index-file.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/** This file is meant to create an index.html file to allow knowing which debug files exists */
+import fs from 'fs';
+
+const htmlFilesLinks = fs.readdirSync('test/debug-pages')
+    .filter(f => f.endsWith('.html'))
+    .map(f => `    <a href='debug-pages/${f}'>${f}</a><br/>`)
```
Comment (HarelM) [inline build/generate-debug-index-file.ts:6]: @birkskyum are you sure about this? Should be either relative to index file or absolute. Witch is neither...?

Comment (birkskyum) [reply build/generate-debug-index-file.ts]: You're right... now it's double 'debug-pages'. Don't know why it seemed to work before.

Comment (birkskyum) [reply build/generate-debug-index-file.ts]: You're right - I reverted my change.

**Comment by HarelM** [comment]
@wipfli you need to approve otherwise this can't be merged...

**Review by wipfli — APPROVED** [review]
Oh, thanks for the ping

**Pull request merged**